src/interfaces/swig/SGBase.i

```diff
@@ -508,45 +486,17 @@ namespace shogun
 #else
             PyString_AsStringAndSize(py_str, &str, &len);
 #endif
-
-            char* fname=tmpnam(NULL);
-            FILE* tmpf=fopen(fname, "w");;
-            size_t total = fwrite(str, (size_t) 1, (size_t) len, tmpf);
-            fclose(tmpf);
-            ASSERT(total==len);
-
-            CSerializableFile* fstream=NULL;
-#ifdef HAVE_HDF5
+            io::CDeserializer* deser = nullptr;
             if (pickle_ascii)
-                fstream = new CSerializableAsciiFile(fname, 'r');
+                deser = new io::CJsonDeserializer();
             else
-            {
-                try
-                {
-                    fstream = new CSerializableHdf5File(fname, 'r');
-                }
-                catch (ShogunException& e)
-                {
-                    fstream = new CSerializableAsciiFile(fname, 'r');
-                }
-            }
-#else
-            try
-            {
-                fstream = new CSerializableAsciiFile(fname, 'r');
-            }
-            catch (ShogunException& e)
-            {
-                    SG_SERROR("File contains an HDF5 stream but " \
-                            "Shogun was not compiled with HDF5" \
-                            " support! -  cannot load file %s." \
-                            " (exception was %s)", e.what());
-            }
-#endif
-            $self->load_serializable(fstream);
-            fstream->close();
-            delete fstream;
-            unlink(fname);
+                deser = new io::CBitseryDeserializer();
+
+            auto byte_input_stream = some<io::CByteArrayInputStream>(str, len);
+            deser->attach(byte_input_stream);
+            auto obj = deser->read();
+            SG_UNREF(deser);
+            $self = obj.get();
```

Comment:
@lisitsyn we have a problem with the new serialization API. both in case here for python (pickle) and java externalizable things are operating over `this`... although copy ctor is not defined for SGObject (might fix it? :S), currently this is not gonna work as expected.